crates/supervisor/storage/src/providers/log_provider.rs

```diff
@@ -250,40 +287,114 @@ mod tests {
         }
     }
 
+    /// Sets up a new temp DB
+    fn setup_db() -> DatabaseEnv {
+        let temp_dir = TempDir::new().expect("Could not create temp dir");
+        init_db_for::<_, Tables>(temp_dir.path(), DatabaseArguments::default())
+            .expect("Failed to init database")
+    }
+
+    /// Helper to initialize database in a new transaction, committing if successful.
+    fn initialize_db(db: &DatabaseEnv, block: &BlockInfo) -> Result<(), StorageError> {
+        let tx = db.tx_mut().expect("Could not get mutable tx");
+        let provider = LogProvider::new(&tx);
+        let res = provider.initialise(*block);
+        if res.is_ok() {
+            tx.commit().expect("Failed to commit transaction");
+        } else {
+            tx.abort();
+        }
+        res
+    }
+
+    /// Helper to insert a pair in a new transaction, committing if successful.
+    fn insert_block_logs(
+        db: &DatabaseEnv,
+        block: &BlockInfo,
+        logs: Vec<Log>,
+    ) -> Result<(), StorageError> {
+        let tx = db.tx_mut().expect("Could not get mutable tx");
+        let provider = LogProvider::new(&tx);
+        let res = provider.store_block_logs(block, logs);
+        if res.is_ok() {
+            tx.commit().expect("Failed to commit transaction");
```

Comment:
same here, are we sure we want the node to be killed in production? how about just panicking in debug mode with `debug_assert` and otherwise returning error?

Reply:
This is part of unit testing